docs/api/vega-wallet/reference/browser-extension/javascript-api.md

```diff
@@ -0,0 +1,260 @@
+---
+title: JavaScript API
+hide_title: false
+vega_network: TESTNET
+sidebar_position: 2
+---
+import Topic from '/docs/topics/_topic-wallet.mdx'
+
+<Topic />
+
+## Vega Browser Wallet
+
+dApps and websites can communicate with the Vega browser wallet over 
+JSON-RPC 2.0, however the browser extension also provides
+a high-level Javascript API, injected into web pages.
+
+### Detecting the extension
+
+The Vega browser wallet injects methods on the global `vega` object,
+hence the presence of the extension can be detected by checking that
+this object is defined:
+
+```js
+if (globalThis.vega != null) {
+    // Vega is installed
+} else {
+    // Prompt the user to install the extension
+}
+```
+
+### Connecting to the extension
+
+To start communicating with the extension, a dApp must first request
+a connection:
+
+```js
+try {
+    await vega.connectWallet()
+} catch (err) {
+    // The user potentially rejected the connection    
+}
+```
+
+The first time a dApp tries to connect to a wallet, the user will be
+prompted to approve the connection and select which keys they will
+allow the dApp to see. Only appoved keys will be available to the dApp.
+
+Subsequent connect attempts will resolve automatically, unless the user
+has removed the connection in the meantime.
+
+Once a dApp is finished talking to the wallet it can call 
+`await vega.disconnectWallet()`.
+
+### Detecting the network in use
+
+The Vega browser wallet allows users to have multiple network
+configurations installed at the same time, to allow switching between
+different entry points into the network or different networks entirely 
+(eg. `mainnet`, `fairground`, etc.)
+
+A dApp can detect the network in use by checking the `chainID` in use:
+
+```js
+const { chainID } = await vega.getChainId()
+```
+
+Note that `chainID` is not an indefinate constant but may change with
```

Comment:
nice

Reply:
Sorry to be that guy but should it be 'indefinite'?